spec/controllers/saml_idp_controller_spec.rb

```diff
@@ -211,6 +211,29 @@
       end
     end
 
+    context 'service provider is valid' do
+      before do
```

Comment:
Oh interesting. That is pretty unusual, in my experience. I don't see the benefit of a `before` block that is not for shared setup, since it behaves exactly the same as just putting that same code at the beginning of the test itself?